### DISCUSSION TITLE
Hot refresh not working: How to debug this problem?

### DISCUSSION THREAD

**Discussion by JoseFMP** [vercel/next.js#17968] [Help]
I was using Next.JS with hot refresh since Next 9.4.X and it was working fine. I updated to 9.5 and still working fine.
I do not remember to have changed anything in the configuration of my app, but suddenly it stopped working.

It means, now when I run `yarn next` the dev server starts properly but once I am using the app in the browser, if I make changes to the source code it will update in the browser as usually but suddenly stops working, i.e. suddenly it does not update the app in the browser any more. No matter what I do, no changes are reflected in the web browser.

This is very frustrating since I have been using Fast Refresh for months with this app and it always worked flawlessly.

Now, I do not expect here an answer on why my Fast Refresh is not working. This is because the react app is quite big and I cannot just really drop the whole app here (I do not believe anyone would have the time to look at it and also is proprietary software).

Question here is: Where should I look at? I did not update any packages, did not change really any configuration anywhere and suddenly one morning stopped working as usual. Now in NextJS 9.5.5 but tried with prior versions that were working properly and no difference.

**Comment by JoseFMP** [comment]
Seems the problem was that I had installed in the same URL path (http://localhost:3000) a service worker from another completely different NextJS project. I unregistered it and restarted the server and now seems to work. I'll report if it keeps working for a few days.

**Comment by daraclare** [comment]
I have the same issue, but no service workers, would be great to hear if this was something else.
